### ISSUE TITLE
Typescript parser removes _wrong_ decorators

### ISSUE THREAD

**Issue by CiGit** [prettier/prettier#5823]
**Prettier 1.16.3**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEABAJnMAKAlAHSkigGcYACAQ3IF5yBGQwjLQgMwFciYBLacgEZ5gAXyZQWYQmAA2lEiXJhghcuUmryAW2FioIkABoQEAA69oJZKEoAnWxADuABTsIrKSgDcIPdEZABW0owAGs4GABlUxCeKABzZBhbDjhjAAsYLRkAdXSeeBIYsDhI9wKeLwKAT2RwBQC4kjhbGGdg+K1KZDZKGWbjACsSAA8AIWCwiMjKLTgAGTi4Hr6BkGGRyLj4mTgARQ4IeBX+tJAY22bbOphq0zgSMFsecwDTZ9gcvxh05AAOAAMxneEGaOWCpjq7weLS8y2MtjgAEcODxEe1KJ1ukheqdjM0tDwkikziRtrsDkdljjVmcYJQBF90D9kAAmYzJSg8GTbADCEC0XTqUGg8JAHGaABUGR5cc0RCIgA)
```sh
--parser typescript
```

**Input:**
```tsx
@dec()
const a = 1

@dec
function b(){}

@dec
class c{
  @dec
  m(){}
}
```

**Output:**
```tsx
const a = 1;

function b() {}

@dec
class c {
  @dec
  m() {}
}

```

**Expected behavior:**

I would expect prettier to fail instead of silently removing those wrong decorators.

**Comment by ikatyang** [comment]
Tracked in typescript-eslint/typescript-eslint#185

**Comment by CiGit** [comment]
Thanks @ikatyang

**Comment by duailibe** [comment]
This was fixed in 1.17

Relevant PR #6027
